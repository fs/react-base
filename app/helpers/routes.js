import pathToRegexp from 'path-to-regexp';

export const routes = {
  home: '/',
  about: '/about',
  profile: '/profile',
  signin: '/signin',
  signup: '/signup',
  aboutExtended: '/about/extended/:id',
};

export const paths = {};

Object.keys(routes)
  .forEach((routeName) => {
    paths[routeName] = pathToRegexp.compile(routes[routeName]);
  });

export default routes;
