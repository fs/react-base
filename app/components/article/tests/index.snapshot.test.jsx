import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter, Switch, Route } from 'react-router';
import { routes, paths } from 'helpers/routes';
import Article from '../';

describe('Article', () => {
  const renderComponent = () => renderer.create((
    <MemoryRouter initialEntries={[paths.aboutExtended({ id: '1' })]} >
      <Switch>
        <Route
          exact
          path={routes.aboutExtended}
          component={Article}
        />
      </Switch>
    </MemoryRouter>
  ));

  it('renders correctly', () => {
    expect(renderComponent().toJSON()).toMatchSnapshot();
  });
});
