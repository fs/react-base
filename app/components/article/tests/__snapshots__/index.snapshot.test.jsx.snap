// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article renders correctly 1`] = `
<article
  dangerouslySetInnerHTML={
    Object {
      "__html": "<br/> <h3>Why React? Edit on GitHub</h3> <p> React is a JavaScript library for creating user interfaces by Facebook and Instagram. Many people choose to think of React as the V in MVC. We built React to solve one problem: building large applications with data that changes over time. </p> <h3>Simple</h3> <p> Simply express how your app should look at any given point in time, and React will automatically manage all UI updates when your underlying data changes. </p> <h3>Declarative</h3> <p> When the data changes, React conceptually hits the \\"refresh\\" button, and knows to only update the changed parts. </p> <h3>Build Composable Components</h3> <p> React is all about building reusable components. In fact, with React the only thing you do is build components. Since they're so encapsulated, components make code reuse, testing, and separation of concerns easy. </p> <h3>Give It Five Minutes</h3> <p> React challenges a lot of conventional wisdom, and at first glance some of the ideas may seem crazy. Give it five minutes while reading this guide; those crazy ideas have worked for building thousands of components both inside and outside of Facebook and Instagram. </p> <br/> <b>article id: 1</b>",
    }
  }
/>
`;
